imod/mf6/boundary_condition.py

```
@@ -298,19 +338,107 @@ def aggregate_layers(self) -> GridDataDict:
+    ) -> Self:
+        """
+        Reallocates topsystem data across layers and create new package with it.
+        Aggregate data to planar data first, by taking either the mean for state
+        variables (e.g. river stage), or the sum for fluxes and the
+        conductance. Consequently allocate and distribute the planar data to the
+        provided model layer schematization.
+
+        Parameters
+        ----------
+        dis : StructuredDiscretization | VerticesDiscretization
+            The discretization of the model to which the data should be
+            reallocated.
+        npf : NodePropertyFlow, optional
+            The node property flow package of the model to which the conductance
+            should be distributed (if applicable). Required for packages with a
+            conductance variable.
+        allocation_option : ALLOCATION_OPTION, optional
+            The allocation option to use for the reallocation. If None, the
+            default allocation option is taken from
+            :class:`imod.prepare.SimulationAllocationOptions`.
+        distributing_option : DISTRIBUTING_OPTION, optional
+            The distributing option to use for the reallocation. Required for
+            packages with a conductance variable. If None, the default is taken
+            from :class:`imod.prepare.SimulationDistributingOptions`.
+
+        Returns
+        -------
+        BoundaryCondition
+            A new instance of the boundary condition class with the reallocated
+            data. The original instance remains unchanged.
+        """
+        allocation_option, distributing_option = _handle_reallocation_options(
+            self._pkg_id, allocation_option, distributing_option
+        )
+        # Aggregate data to planar data first
+        planar_data = self.aggregate_layers(self.dataset)
+        if "conductance" in self.dataset.data_vars:
+            if npf is None:
```

You can move this check to the start of this method. That way you can fail fast and don't have to call aggregate_layers

```
...
 BoundaryCondition
            A new instance of the boundary condition class with the reallocated
            data. The original instance remains unchanged.
        """
if "conductance" in self.dataset.data_vars and npf is None:
  raise ValueError(
                    "NodePropertyFlow must be provided "
                    "for packages with conductance variable."
                )

 # Aggregate data to planar data first
  planar_data = self.aggregate_layers(self.dataset)

...
```

---

Good point, thanks!